packages/hathor-rpc-handler/src/rpcMethods/getAddress.ts

```diff
@@ -20,25 +20,38 @@ import { NotImplementedError, PromptRejectedError, InvalidParamsError } from '..
 import { validateNetwork } from '../helpers';
 import type { AddressInfoObject } from '@hathor/wallet-lib/lib/wallet/types';
 
-const getAddressSchema = z.object({
-  type: z.enum(['first_empty', 'full_path', 'index', 'client']),
-  index: z.number().int().nonnegative().optional(),
-  full_path: z.string().min(1).optional(),
+const baseSchema = {
   network: z.string().min(1),
-}).refine(
-  (data) => {
-    if (data.type === 'index') {
-      return data.index !== undefined;
-    }
-    if (data.type === 'full_path') {
-      return data.full_path !== undefined;
-    }
-    return true;
-  },
-  {
-    message: "index is required when type is 'index', full_path is required when type is 'full_path'",
+};
+
+const getAddressSchema = z.discriminatedUnion("type", [
+  z.object({
+    type: z.literal('first_empty'),
+    ...baseSchema,
```

Comment:
Since we have a baseSchema we could use [extend](https://zod.dev/?id=extend) or [merge](https://zod.dev/?id=merge) which would be something like `baseSchema.extend({ type: z.literal('first_empty') })`

Reply:
Looks better, thanks